### DISCUSSION TITLE
Debug zx script in vscode

### DISCUSSION THREAD

**Discussion by pguardiario** [google/zx#267] [General]
Is it possible to debug a zx script in vscode? Can anyone post a launch.json for reference?

**Comment by domiSchenk** [comment]
What works for me is using the JavaScript Debug Terminal.
then i just run zx <file>.mjs

**Comment by tbxark** [comment]
- Create a node project and add `zx` to the dependencies.
- Add `import "zx/globals"`  in the script when debugging;
- Then you can debug like a normal node project.

example: https://github.com/TBXark/github-backup/blob/master/package.json

**Comment by iamsarvagyaa** [comment]
you can use JavaScript Debug Terminal that works :)

**Comment by Mellbourn** [comment]
I managed to get it working by starting with the standard `Node.js` configuration, and then setting the `runtimeExcecutable` to `zx`, and the `program` to the path of my zx script.

So this is my `launch.json`:

```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "node",
            "request": "launch",
            "name": "zx launcher",
            "skipFiles": [
                "<node_internals>/**"
            ],
            "runtimeExecutable": "${env:HOME}/.asdf/shims/zx",
            "program": "${workspaceFolder}/../../bin/cleanup-branches.spec.mjs"
        }
    ]
}
```
